internal/pkg/build/remotebuilder/remotebuilder.go

```diff
@@ -84,159 +80,87 @@ func (rb *RemoteBuilder) Build(ctx context.Context) (err error) {
-				err = errors.Wrap(err, "failed to pull image file")
-				sylog.Warningf("%v", err)
-				return err
-			}
-		}
-	}
-
-	return nil
-}
-
-// streamOutput attaches via websocket and streams output to the console
-func (rb *RemoteBuilder) streamOutput(ctx context.Context, url string) (err error) {
-	h := http.Header{}
-	rb.setAuthHeader(h)
-	h.Set("User-Agent", useragent.Value())
-
-	c, resp, err := websocket.DefaultDialer.Dial(url, h)
+	// We're doing an attached build, stream output and then download the resulting file
+	var outputLogger stdoutLogger
+	err = rb.BuildClient.GetOutput(ctx, bi.ID, outputLogger)
 	if err != nil {
-		sylog.Debugf("websocket dial err - %s, partial response: %+v", err, resp)
+		err = errors.Wrap(err, "failed to stream output from remote build service")
+		sylog.Warningf("%v", err)
 		return err
 	}
-	defer c.Close()
-
-	for {
-		// Check if context has expired
-		select {
-		case <-ctx.Done():
-			return ctx.Err()
-		default:
-		}
-
-		// Read from websocket
-		mt, msg, err := c.ReadMessage()
-		if err != nil {
-			if websocket.IsCloseError(err, websocket.CloseNormalClosure) {
-				return nil
-			}
-			sylog.Debugf("websocket read message err - %s", err)
-			return err
-		}
 
-		// Print to terminal
-		switch mt {
-		case websocket.TextMessage:
-			fmt.Printf("%s", msg)
-		case websocket.BinaryMessage:
-			fmt.Print("Ignoring binary message")
-		}
-	}
-}
-
-// doBuildRequest creates a new build on a Remote Build Service
-func (rb *RemoteBuilder) doBuildRequest(ctx context.Context, d types.Definition, libraryRef string) (rd types.ResponseData, err error) {
-	if libraryRef != "" && !client.IsLibraryPushRef(libraryRef) {
-		err = fmt.Errorf("invalid library reference: %v", rb.ImagePath)
+	// Get build status
+	bi, err = rb.BuildClient.GetStatus(ctx, bi.ID)
+	if err != nil {
+		err = errors.Wrap(err, "failed to get status from remote build service")
 		sylog.Warningf("%v", err)
-		return types.ResponseData{}, err
+		return err
 	}
 
-	b, err := json.Marshal(types.RequestData{
-		Definition: d,
-		LibraryRef: libraryRef,
-		LibraryURL: rb.LibraryURL,
-	})
-	if err != nil {
-		return
+	// Do not try to download image if not complete or image size is 0
+	if !bi.IsComplete {
+		return errors.New("build has not completed")
 	}
-
-	req, err := http.NewRequest(http.MethodPost, rb.BuilderURL.String()+"/v1/build", bytes.NewReader(b))
-	if err != nil {
-		return
+	if bi.ImageSize <= 0 {
+		return errors.New("build image size <= 0")
 	}
-	req = req.WithContext(ctx)
-	rb.setAuthHeader(req.Header)
-	req.Header.Set("User-Agent", useragent.Value())
-	req.Header.Set("Content-Type", "application/json")
-	sylog.Debugf("Sending build request to %s", req.URL.String())
 
-	res, err := rb.Client.Do(req)
-	if err != nil {
-		return
+	// If image destination is local file, pull image.
+	if !strings.HasPrefix(rb.ImagePath, "library://") {
+		err = client.DownloadImage(rb.ImagePath, bi.LibraryRef, bi.LibraryURL, rb.Force, rb.AuthToken)
```

Comment:
`err` is a new variable in this statement:

```suggestion
		err := client.DownloadImage(rb.ImagePath, bi.LibraryRef, bi.LibraryURL, rb.Force, rb.AuthToken)
```

Reply:
It can be a new variable, but it can also just re-use the err that is already defined. This is a pretty common pattern in Go, but if you feel strongly it should be done a different way I can use your suggestion.

Reply:
I dont feel strongly about it. Imo I would add the `:`, but maybe @mem or someone else has a opinion?